Integrate new protection tools into api's offer & trade services

ghubstan opened the pull request:
- Injected `OfferFilter` into `CoreOffersService` and `CoreTradesService`.
The filter constrains `getoffer(s)` results to offers an api user can take, as the first part of ongoing protection tools / api integration.

- Created a new `CoreContext` singleton.
Initially, lets `Core*Services` know if the user is using the api -- `isApiUser=true` if the current thread's name is "BisqDaemonMain", or the name contains "grpc".  We cannot check `BisqDaemonMain.class.getSimpleName()` because `:core` cannot have a circular dependency on `:daemon`, but there is probably a better way to do this than depending on the thread name set in `BisqDaemonMain#configUserThread()`.  We do this anticipating future `:desktop` dependencies on the core api, and we don't pass hardcoded `isApiUser=true` params to lower level domain objects.  

- Added `@Singleton` annotation to all Core*Service classes.
      
Related to https://github.com/bisq-network/bisq/pull/5053.

PR https://github.com/bisq-network/bisq/pull/5063 should be reviewed and merged before this one.

ghubstan committed:
Fix two CLI option parsing bugs

- Fix bug locating index of method name in String[] args.

- Make createoffer method's security-deposit param format consistent with UI's.

  When creating an offer, the CLI should take "15.0", not "0.15" as
  a 15% security deposit.  This is consistent with the UI, and the
  CLI's mkt-price-margin input format.

ghubstan committed:
Add @Singleton annotation

ghubstan committed:
Integrate new protection tools into api's offer & trade services

- Injected OfferFilter into CoreOffersService and CoreTradesService.

  The filter constrains 'getoffer(s)' results to offers an api user
  can take, as the first part of ongoing protection tools / api
  integration.

- Created a new CoreContext singleton.

  Initially, lets Core*Services know if the user is using the api --
  isApiUser=true if the current thread's name is "BisqDaemonMain" or
  the name contains "grpc".  We do this anticipating future :desktop
  dependencies on the core api, and we don't pass hardcoded
  isApiUser=true params to lower level domain objects.

  We cannot check BisqDaemonMain.class.getSimpleName() because :core
  cannot have a circular dependency on :daemon, but there is probably a
  better way to do this than depending on the thread name set in
  BisqDaemonMain#configUserThread().

- Added @Singleton annotation to all Core*Service classes.

sqrrm commented on core/src/main/java/bisq/core/api/CoreContext.java:
```diff
@@ -0,0 +1,39 @@
+
+    public boolean isApiUser() {
+        String threadName = currentThread().getName();
+        return threadName.equals("BisqDaemonMain") || threadName.contains("grpc");
```
I find this ugly. There must be some easier and not quite as ugly way to do this.

ghubstan replied:
Resolved by commit 0e779a4

sqrrm reviewed (commented):
It hurts my soul somewhat with that thread parsing to figure out context. I think we should try to think a bit more before merging something like that.

ghubstan committed:
Inject CoreContext into server, then set isApiUser=true

Do not set isApiUser=true based on hardcoded thread name ;-(

For requested change
https://github.com/bisq-network/bisq/pull/5065#pullrequestreview-567708569

sqrrm reviewed (approved):
utACK

I much prefer this solution
